cmd/controller-manager/app/controller-manager.go

```diff
@@ -239,6 +317,8 @@ func setOptionsByFederationConfig(opts *options.Options) {
 	opts.ClusterHealthCheckConfig.FailureThreshold = spec.ClusterHealthCheck.FailureThreshold
 	opts.ClusterHealthCheckConfig.SuccessThreshold = spec.ClusterHealthCheck.SuccessThreshold
 
+	updateFederationConfig(opts.Config.KubeConfig, fedConfig)
```

Comment:
As proposed, `FederationConfig` will only be written the API if it already existed in the API or was provided by command line.  It should probably always be written to ensure that `kubefed2` can rely on it existing.

Reply:
Yes. No matter specified by command line `--federation-config` or exist `FederationConfig`, we will create/update `FederationConfig` for kubefed2 use.

Reply:
It appears that if `getFederationConfig` returns nil because `FederationConfig` is not provided by flag or does not exist in the API, this function will exit on line 303 without setting defaults and without writing them back to the API.

Reply:
I assumed that configuration is the hard dependecy for federation before. Now I realize it does not with your comments. 

If there is no configuration at all, we can make all by defaults and create `FederationConfig`.